### PR TITLE
[test] Added disable_checkpointing unit test

### LINKED ISSUE #778: [test] need a test to cover the checkpoint disabling API
see issue #771

### PR DESCRIPTION
**Pull request by JARACH-209**
## What does this PR do?
Fixes #778 .
https://github.com/facebookresearch/fairscale/issues/778#issue-988653326

## Before submitting

- [x] Did you have fun?
  - Make sure you had fun coding 🙃
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
  - [ ] N/A
- [ ] Did you make sure to update the docs?
  - [ ] N/A
- [x] Did you write any new necessary tests?
  - [ ] N/A
- [ ] Did you update the [changelog](https://github.com/facebookresearch/fairscale/blob/master/CHANGELOG.md)? (if needed)
  - [ ] N/A

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by JARACH-209** [comment]
@min-xu-ai  Please review and suggest necessary changes if rquired.

File: tests/nn/checkpoint/test_checkpoint_activations.py [hunk 1]
```diff
@@ -303,3 +303,39 @@ def forward(self, x):
...
+
+
+def test_checkpoint_disabling():
+    """ Test to check new disable_checkpoint() API added to checkpoint_wrapper."""
+
+    class TestModel(nn.Module):
+        def __init__(self):
+            super().__init__()
+            self.cnt = 0
+            self.conv = nn.Linear(2, 2)
```
Comment (min-xu-ai) [inline tests/nn/checkpoint/test_checkpoint_activations.py:315]:
```suggestion
            self.linear = nn.Linear(2, 2)
```

File: tests/nn/checkpoint/test_checkpoint_activations.py [hunk 2]
```diff
@@ -303,3 +303,39 @@ def forward(self, x):
...
+
+        def forward(self, x):
+            self.cnt += 1
+            y = []
+            for i in x:
+                y.append(self.conv(i))
```
Comment (min-xu-ai) [inline tests/nn/checkpoint/test_checkpoint_activations.py:321]:
```suggestion
                y.append(self.linear(i))
```

File: tests/nn/checkpoint/test_checkpoint_activations.py [hunk 3]
```diff
@@ -303,3 +303,39 @@ def forward(self, x):
...
+            return y
+
+    x = in_data1 = torch.rand(4, 2)
+    model1 = checkpoint_wrapper((TestModel()))
```
Comment (min-xu-ai) [inline tests/nn/checkpoint/test_checkpoint_activations.py:325]:
```suggestion
    model1 = checkpoint_wrapper(TestModel())
```

**Review by min-xu-ai — APPROVED** [review]
I really like this test! Thanks!

I added @anj-s for her review as well.

**Comment by JARACH-209** [comment]
@min-xu-ai  I have done the suggested clean-up. The tests were functional and passing. Thanks!

**Review by anj-s — APPROVED** [review]

**Pull request merged**